### PR TITLE
tikv: enable async-io by setting `raftstore.store-io-pool-size == 1` as default.

### PR DESCRIPTION
**Pull request by LykxSassinator**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): issue: https://github.com/tikv/tikv/issues/16614, tikv-pr: https://github.com/tikv/tikv/pull/16615

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by qiancai** [comment]
Hi @LykxSassinator, may I know whether this change is introduced starting from TiDB v8.0?

**Comment by LykxSassinator** [comment]
> Hi @LykxSassinator, may I know whether this change is introduced starting from TiDB v8.0?

Yep

**Review by qiancai — APPROVED** [review]

**Review by glorv — APPROVED** [review]

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -989,7 +989,7 @@ raftstore 相关的配置项。
 ### `store-io-pool-size` <span class="version-mark">从 v5.3.0 版本开始引入</span>
 
 + 表示处理 Raft I/O 任务的线程池中线程的数量，即 StoreWriter 线程池的大小。调整该线程池的大小时，请参考 [TiKV 线程池调优](/tune-tikv-thread-performance.md#tikv-线程池调优)。
-+ 默认值：0
++ 默认值：1
```
Comment (qiancai) [inline tikv-configuration-file.md:992]:
```suggestion
+ 默认值：1 （对于 TiDB v8.0.0 之前的版本，默认值为 0）
```

File: tikv-configuration-file.md [hunk 2]
```diff
@@ -989,7 +989,7 @@ raftstore 相关的配置项。
 ### `store-io-pool-size` <span class="version-mark">从 v5.3.0 版本开始引入</span>
 
 + 表示处理 Raft I/O 任务的线程池中线程的数量，即 StoreWriter 线程池的大小。调整该线程池的大小时，请参考 [TiKV 线程池调优](/tune-tikv-thread-performance.md#tikv-线程池调优)。
-+ 默认值：0
++ 默认值：1 （对于 TiDB v8.0.0 之前的版本，默认值为 0）
```
Comment (qiancai) [inline tikv-configuration-file.md:992]:
```suggestion
+ 默认值：1（对于 TiDB v8.0.0 之前的版本，默认值为 0）
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**